app/helpers/handlers.js

```diff
@@ -23,6 +23,70 @@ const {
   getConsentOptionalData
 } = require('./pageHelpers')
 
+const setGrantsData = (question, request) => {
+  if (question.grantInfo) {
+    const { calculatedGrant, remainingCost } = getGrantValues(getYarValue(request, 'itemsTotalValue'), question.grantInfo)
+    setYarValue(request, 'calculatedGrant', calculatedGrant)
+    setYarValue(request, 'remainingCost', remainingCost)
+  }
+};
+
+const sendContactDetails = async (request) => {
+  try {
+    await senders.sendContactDetails(createMsg.getAllDetails(request, confirmationId), request.yar.id)
+    await gapiService.sendDimensionOrMetrics(request, [ {
+      dimensionOrMetric: gapiService.dimensions.CONFIRMATION,
+      value: confirmationId
+    }, {
+      dimensionOrMetric: gapiService.dimensions.FINALSCORE,
+      value: getYarValue(request, 'current-score')
+    },
+    {
+      dimensionOrMetric: gapiService.metrics.CONFIRMATION,
+      value: 'TIME'
+    }
+    ])
+    console.log('Confirmation event sent') // no sonar
+  } catch (err) {
+    console.log('ERROR: ', err)
+  }
+}
+
+const setTitle = async (title, question, request) => {
+  if (title) {
+    return {
+      ...question,
+      title: title.replace(SELECT_VARIABLE_TO_REPLACE, (_ignore, additionalYarKeyName) => (
+        formatUKCurrency(getYarValue(request, additionalYarKeyName) || 0)
+      ))
+    }
+  }
+}
+const processGA = async (question, request, confirmationId) => {
+  if (question.ga) {
+    await gapiService.processGA(request, question.ga, confirmationId)
+  }
+}
+
+const addConsentOptionalData = async (url, request) => {
+  if (url === 'confirm') {
+    const consentOptional = getYarValue(request, 'consentOptional')
+    return getConsentOptionalData(consentOptional)
+  }
+}
+
+const addConditionalLabelData = async (question, type, request, condHTML) => {
```

Comment:
yarKey is also mentioned in this function, yet it does not seem to be declared. Is this something that also needs to be passed as a parameter?

Reply:
Have added parameter